### PR TITLE
Reworked HISAT2 input parameters.

### PR DESCRIPTION
**Pull request by davebx**
- Got rid of the collection type conditional, since the correct
behavior is to consume a list of pairs and produce a list of alignments,
or a simple pair and produce a single file. This is handled natively by `collection_type="paired"`
- Normalized the XML.

### CODE REVIEW

**Comment by nsoranzo** [comment]
> Normalized the XML

Can you revert that? We have a [coding style](https://galaxy-iuc-standards.readthedocs.io/en/latest/best_practices/tool_xml.html#coding-style) with the suggested attribute order.

File: tools/hisat2/hisat2.xml [hunk 1]
```diff
@@ -1,21 +1,20 @@
-<?xml version="1.0"?>
-<tool id="hisat2" name="HISAT2" version="2.0.5.1" profile="17.01">
+<tool id="hisat2" name="HISAT2" profile="17.01" version="2.0.5.1">
```
Comment (nsoranzo) [inline tools/hisat2/hisat2.xml]: Increase version number.

File: tools/hisat2/hisat2_macros.xml [hunk 2]
```diff
@@ -46,17 +51,15 @@
...
             </when>
         </conditional>
     </xml>
     <xml name="paired_end_output">
-        <param name="unaligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write unaligned reads (in fastq format) to separate file(s)" argument="--un-conc / --un-conc-gz" />
-        <param name="aligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write aligned reads (in fastq format) to separate file(s)" argument="--al-conc / --al-conc-gz" />
+        <param argument="--un-conc / --un-conc-gz" checked="False" falsevalue="false" label="Write unaligned reads (in fastq format) to separate file(s)" name="unaligned_file" truevalue="true" type="boolean" />
```
Comment (bgruening) [inline tools/hisat2/hisat2_macros.xml]: This argument does not work, can you just use one, e.g. `--un-conc-gz`.
I'm also not sure about the parameter ordering, sometimes argument is the last, here it is the first.

File: tools/hisat2/hisat2_macros.xml [hunk 3]
```diff
@@ -12,29 +12,34 @@
...
+                    </when>
+                </conditional>
                 <expand macro="paired_end_conditional" />
                 <expand macro="paired_end_output" />
             </when>
             <when value="single">
                 <param format="@FTYPE@" label="Reads" name="reads" type="data" />
-                <param name="unaligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write unaligned reads (in fastq format) to separate file(s)" argument="--un / --un-gz" />
-                <param name="aligned_file" type="boolean" truevalue="true" falsevalue="false" checked="False" label="Write aligned reads (in fastq format) to separate file(s)" argument="--al / --al-gz" />
+                <param argument="--un / --un-gz" checked="False" falsevalue="false" label="Write unaligned reads (in fastq format) to separate file(s)" name="unaligned_file" truevalue="true" type="boolean" />
```
Comment (bgruening) [inline tools/hisat2/hisat2_macros.xml]: please just use one argument in argument .)

**Comment by nsoranzo** [comment]
Anyone wants to give a final review of this one?

**Comment by shiltemann** [comment]
When I try to run this on a pair or list of pair I get the following error (before tool is even executed) 

```
The server could not complete the request. Please contact the Galaxy Team if this error 
persists. Error executing tool: argument of type 'DatasetCollectionElement' is not iterable

{
    "tool_id": "hisat2",
    "tool_version": "2.0.5.2",
    "inputs": {
        "input_format|input_format_selector": "fastq",
        "input_format|paired|paired_selector": "paired",
        "input_format|paired|collection|collection_selector": "collection",
        "input_format|paired|collection|reads": {
            "values": [
                {
                    "hid": 15,
                    "id": "5729865256bc2525",
                    "map_over_type": "paired",
                    "name": "pairlist",
                    "src": "hdca",
                    "tags": []
                }
            ],
            "batch": true
        },
        "input_format|paired|paired_end_options|paired_end_options_selector": "defaults",
        "input_format|paired|unaligned_file": "true",
        "input_format|paired|aligned_file": "true",
        "reference_genome|reference_genome_source": "history",
        "reference_genome|history_item": {
            "values": [
                {
                    "hid": 13,
                    "id": "80b8022ff3f677b7",
                    "keep": false,
                    "name": "phiX.fa",
                    "src": "hda",
                    "tags": []
                }
            ],
            "batch": false
        },
        "max_primary": "",
        "max_seeds": "",
        "secondary": "false",
        "alignment_options|alignment_options_selector": "defaults",
        "input_options|input_options_selector": "defaults",
        "scoring_options|scoring_options_selector": "defaults",
        "spliced_options|spliced_options_selector": "defaults",
        "paired_options|paired_options_selector": "defaults"
    }
}
```

**Comment by davebx** [comment]
@shiltemann I'm not able to reproduce this error locally, can you confirm that it's still broken?

**Comment by shiltemann** [comment]
seems to work for me again too @davebx, must have been some misconfiguration at my end orso

**Comment by nsoranzo** [comment]
Thanks a lot @davebx and @shiltemann!

**Pull request merged**